app/assets/javascripts/discourse/components/ignored-user-list.js.es6

```diff
@@ -0,0 +1,29 @@
+import { popupAjaxError } from "discourse/lib/ajax-error";
+import showModal from "discourse/lib/show-modal";
+import User from "discourse/models/user";
+
+export default Ember.Component.extend({
+  item: null,
+  actions: {
+    removeItem(item) {
+      this.set("saved", false);
+      this.get("items").removeObject(item);
+      User.findByUsername(item).then(user => {
+        user
+          .updateNotificationLevel("normal")
+          .catch(popupAjaxError)
+          .finally(() => this.set("saved", true));
+      });
+    },
+    newItem() {
+      const modal = showModal("ignore-duration-with-username", {
```

Comment:
depending on your needs you could rewrite it as:

```
const modal = showModal("ignore-duration-with-username", {
  model: {
    betterModelName: this.get("model"),
    onUserIgnored: username => {
      this.get("items").addObject(username);
    }
  }
});
```

Reply:
I am not sure you can do that with the `showModal`